cookbooks/bcpc-hadoop/recipes/configs.rb

```diff
@@ -62,9 +64,59 @@
 # Install Java
 include_recipe "bcpc-hadoop::java_config"
 include_recipe "java::default"
+include_recipe "java::oracle_jce"
 
 %w{zookeeper}.each do |pkg|
   package pkg do
     action :upgrade
   end
 end
+
+# Create Keytabs (if kerberos is eanbled)
+if node[:bcpc][:hadoop][:enable_kerberos] == true then
+
+  directory "#{node[:bcpc][:hadoop][:kerberos][:keytab][:dir]}" do
+    owner "root"
+    group "root"
+    recursive true
+    mode 0755  
+  end
+
+  # Download and create all keytabs
+  node[:bcpc][:hadoop][:kerberos][:data].each do |srvc, srvdat|
+
+    config_host = srvdat['princhost'] == "_HOST" ?  float_host(node[:hostname]) : srvdat['princhost'].split('.')[0]
+    file "#{node[:bcpc][:hadoop][:kerberos][:keytab][:dir]}/#{srvdat['keytab']}" do
+      action :delete
+      only_if {File.exists?("#{node[:bcpc][:hadoop][:kerberos][:keytab][:dir]}/#{srvdat['keytab']}") && node[:bcpc][:hadoop][:kerberos][:keytab][:recreate] == true}
```

Comment:
Isn't this unnecessary if the guard on line 99 has the `!File.exsists?` removed and we trust Chef to tell if a file needs updated -- or would there be a time we would want to run with an outdated keytab? (It seems we would very quickly rev a user's credentials cache any ways in the loop starting on line 104.)